lib_xua/api/xua_conf_default.h

```diff
@@ -1637,3 +1638,8 @@ enum USBEndpointNumber_Out
 #error CODEC_MASTER with DSD is currently unsupported
 #endif
 
+// When enumerating as bcdUSB 2.01, non-zero iSerialNumber with an empty serialStr doesn't seem to be allowed.
+// If enabling this, ensure that g_strTable.serialStr is set so a non-empty string.
+#ifndef REPORT_USB_SERIAL_NUMBER
```

Comment:
Rather than having the user set this (or forget to) could this be based on something like: 

#if defined(SERIAL_STR) && sizeof(SERIAL_STR) == sizeof("")
#define REPORT_USB_SERIAL_NUMBER 0

..

Reply:
For new defines we should try and stick to the naming convention:

Something to be set by user:
XUA_SOMETHING 

Something we've generated for internal use, not expected to be set by user.
_XUA_SOMETHING_INTERNAL

Reply:
g_serial_str can be set to something other than SERIAL_STR by calling XUA_Endpoint0_setSerialStr(), so I don't think we can depend on SERIAL_STR. Let me try to get rid of REPORT_USB_SERIAL_NUMBER and try to set iSerialNumber to 0 or not depending on whether g_strTable.serialStr is "" or not.